docs/development/testing.md

```diff
@@ -31,27 +31,64 @@ Install [geckodriver](https://github.com/mozilla/geckodriver/releases) and
 [chromedriver](https://sites.google.com/a/chromium.org/chromedriver/downloads)
 and check that they are in your `PATH`.
 
-From the root directory of Pyodide, type on the command line:
+### Running the Python test suite
+
+To run the test suite, run `pytest` from the root directory of Pyodide:
 
 ```bash
 pytest
 ```
 
-#### Playwright
+You can run the tests from a specific file with:
+
+```bash
+pytest path/to/test/file.py
+```
+
+To speed things up you may wish to filter out two of the browsers. Node runs
+tests the fastest so:
+
+```bash
+pytest -k "not chrome and not fire"
+```
+
+is often helpful. Some browsers sometimes produce informative errors than others
+so if you are getting confusing errors it is worth rerunning the test on each
+browser. If you are still confused, try {ref}`manual-testing` which is more
+flexible.
+
+There are 5 test locations that are collected by pytest:
+
+- `src/tests/`: general Pyodide tests and tests running the CPython test suite
+- `pyodide-test-runner/pyodide_test_runner/tests/`: Tests for the testing
+  system.
+- `pyodide-build/pyodide_build/tests/`: tests related to Pyodide build system
+  (do not require selenium to run)
+- `packages/test_common.py`: tests whether all packages can be imported
+  successfully
```

Comment:
```suggestion
- `packages/test_common.py`: common tests for packages.
```
it's not necessarily imports tests, also tests of the meta.yaml in general, and we might add other tests there in the future.